pkg/kubelet/kubelet.go

```diff
@@ -901,6 +901,23 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// If the cloud has zone information, label the node with the zone information
+		zones, ok := kl.cloud.Zones()
```

Comment:
My main concern is that this PR will open up the door for adding more and more node specific labels into the kubelet. 
One compromise could be to explicitly define the limited subset of labels that the kubelet will surface. We need some process for enforcing this though.
The other alternative I can think of is to store all custom labels into a file on the node, which the kubelet can then use. This approach requires labels to be pre-populated though.